README.md

```diff
@@ -30,6 +30,44 @@ videojs('player_id').ready(function () {
 </script>
 ```
 
+## Known Issues
+
+1. Fairplay content cannot be played if `TextTrack`s exist. Beyond embedding them within the manifest, the only way to side load them is after the keys are updated.
+
+  ```js
+  videojs('player_id').ready(function () {
+    var player = this;
+
+    player.on('loadeddata', function () {
+      player.addRemoteTextTrack(...);
+    });
+  });
+  ```
+
+1. New Fairplay content cannot be played once MediaKeys have been attached to the video element (Unless its the same source). To get around this issue we need to clear those keys.
+
+  ```js
+  player.reset();
+
+  player.src(...);
+  ```
+
+1. Combining the issues of #1 and #2. Before you can play *another* source, you must clear out the existing `TextTrack`s.
+
+  ```js
+  player.remoteTextTracks().tracks_.forEach(function (track) {
+    player.removeRemoteTextTrack(track);
+  });
+
+  // the current `TextTrack` implementation operates on events and because of this we have to wait a
+  // tick before executing a reset, otherwise the `TextTrack`s will get picked up in its associated `Tech`.
+  setTimeout(function () {
+    player.reset();
```

Comment:
couldn't you call `player.reset` outside of this setTimeout and then keep `player.src` inside?
The tracks should end up getting removed as part of `player.reset`.

Reply:
I see the problem in the fact that you are registering new event listener with every change of source and the previous one is not removed.
https://github.com/chemoish/videojs-fairplay/blob/5a089488e544468e5b098a167f5d18d424b4740a/src/videojs-fairplay.js#L263

Then I think the reset is not necessary at all and the text tracks should work normally (manual cleanup - true).

Reply:
Makes sense, I'm not doing video stuff at the moment. Do you mind adding a PR is this is effecting you?

Reply:
(But also, I think I wasn't able to update sources without this. That is why I reset and reattached the event if I remember correctly.)